cubano-core/src/main/java/org/concordion/cubano/driver/action/ActionWait.java

```diff
@@ -356,15 +353,17 @@ private int getNextPollingInterval(Instant end) {
             interval = pollingIntervals.get(attempts);
         }
 
-        long endTime = end.toEpochMilli();
-        long currentTime = clock.instant().toEpochMilli();
-        long waitTime = currentTime + TimeUnit.MILLISECONDS.convert(interval, pollingTimeUnit);
-        long stretchTime = waitTime + TimeUnit.MILLISECONDS.convert(interval / 2, pollingTimeUnit);
-
-        // If going above timeout limit bring it back to that limit
-        // If closer than half current interval stretch it out
-        if (waitTime > endTime || stretchTime > endTime) {
-            interval = pollingTimeUnit.convert(endTime - currentTime, TimeUnit.MILLISECONDS) + 1;
+        if (isWaitStyleTimeout()) {
+            long endTime = end.toEpochMilli();
+            long currentTime = clock.instant().toEpochMilli();
+            long waitTime = currentTime + TimeUnit.MILLISECONDS.convert(interval, pollingTimeUnit);
+            long stretchTime = waitTime + TimeUnit.MILLISECONDS.convert(interval / 2, pollingTimeUnit);
```

Comment:
I've updated the comments somewhat, but yes it probably should have some tests around it.  Basically if the wait time for the current polling interval would take it close to the timeout limit then only do the one sleep and final poll rather than sleep, poll, short sleep and final poll.

Reply:
I've added the test "withTimeoutStretchInterval" to illustrate this, hopefully it's not too fragile as I've found differences in running time sensitive tests locally and on travis, even on travis and different java versions.